scilab/modules/scicos_blocks/src/c/cscope.c

```diff
@@ -597,17 +599,64 @@ static void appendData(scicos_block * block, int input, double t, double *data)
 
     /*
      * Update data
+     *
+     * Implements Run-Time Auto scaling of graphs
+     *
      */
     if (sco != NULL)
     {
         for (i = 0; i < block->insz[input]; i++)
         {
-            const double value = data[i];
+            double value = data[i];
+            //printf("value=%lf\n",value);
             setBuffersCoordinates(block, sco->internal.bufferCoordinates[input][i], sco->internal.numberOfPoints, block->ipar[2], t, value);
+            
+            double max_curr_val,prev_max_curr_val;
+            //Get the current maximum value of the axes
+            max_curr_val = block->rpar[2];
+            prev_max_curr_val = max_curr_val;
+            
+            double min_curr_val,prev_min_curr_val;
+            //Get the current minimum value of the axes
+            min_curr_val = block->rpar[1];
+            prev_min_curr_val = min_curr_val;
+            
+            /* If the value to be plotted exceeds the current range, then we update the range. 
+             * We could update the range from current value to the new value i.e. (value + R) 
+             * where R varies from 0 to approx 150. So we have used (value + 100.0) to give the graph good feature.
+             * 
+             * However, the auto-scaling feature implemented is general and will work fine even for 
+             */
+            
+            //If the value to be plotted is greater than the current max, then update the current max            
+            if(value > max_curr_val)
+            {
+            	max_curr_val = value + 10.0;
+            	block->rpar[2] = max_curr_val;
+            }	
+            	
+            //If the value to be plotted is smaller than the current min, then update the current min	
+            if(value < min_curr_val)
+            {
+            	min_curr_val = value - 10.0;
+            	block->rpar[1] = min_curr_val;
+	    }
```

Comment:
write in detail what you want to tell.....

On Mon, Jun 27, 2016 at 3:32 PM, malayaleecoder <notifications@github.com>
wrote:

> In scilab/modules/scicos_blocks/src/c/cscope.c
> <https://github.com/FOSSEE-Internship/scilab/pull/1#discussion_r68551123>:
>
> > +             * However, the auto-scaling feature implemented is general and will work fine even for
> > +             */
> > +
> > +            //If the value to be plotted is greater than the current max, then update the current max
> > +            if(value > max_curr_val)
> > +            {
> > +            	max_curr_val = value + 10.0;
> > +            	block->rpar[2] = max_curr_val;
> > +            }	
> > +            	
> > +            //If the value to be plotted is smaller than the current min, then update the current min	
> > +            if(value < min_curr_val)
> > +            {
> > +            	min_curr_val = value - 10.0;
> > +            	block->rpar[1] = min_curr_val;
> > +	    }
>
> @mittalpulkit <https://github.com/mittalpulkit>
> ^ ditto nit fix
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/FOSSEE-Internship/scilab/pull/1/files/6cbef4bafba7e802b2e5036b9768fdf9ce7f8767#r68551123>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AS-o5RXdRsKSdTBuD5t2oR2Ua2rx0h9Fks5qP5_KgaJpZM4I-y8q>
> .
>